packages/vue/src/utils/create-ts-config.ts

```diff
@@ -1,7 +1,72 @@
 import { Tree, updateJson, writeJson } from '@nx/devkit';
 import * as shared from '@nx/js/src/utils/typescript/create-ts-config';
+import { isUsingTsSolutionSetup } from '@nx/js/src/utils/typescript/ts-solution-setup';
 
 export function createTsConfig(
+  host: Tree,
+  projectRoot: string,
+  type: 'app' | 'lib',
+  options: {
+    strict?: boolean;
+    style?: string;
+    bundler?: string;
+    rootProject?: boolean;
+    unitTestRunner?: string;
+  },
+  relativePathToRootTsConfig: string
+) {
+  if (isUsingTsSolutionSetup(host)) {
+    createTsConfigForTsSolution(host, projectRoot, type, options);
+  } else {
+    createTsConfigForNonTsSolution(
+      host,
+      projectRoot,
+      type,
+      options,
+      relativePathToRootTsConfig
+    );
+  }
+}
+
+export function createTsConfigForTsSolution(
+  host: Tree,
+  projectRoot: string,
+  type: 'app' | 'lib',
+  options: {
+    strict?: boolean;
+    style?: string;
+    rootProject?: boolean;
+    unitTestRunner?: string;
+  }
+) {
+  const json = {
+    files: [],
+    include: [],
+    references: [
+      {
+        path: type === 'app' ? './tsconfig.app.json' : './tsconfig.lib.json',
+      },
+    ],
+  } as any;
```

Comment:
We need to extend from the `tsconfig.base.json`, or is that done somewhere else?